src/node_crypto.cc

```diff
@@ -6113,11 +6113,14 @@ void GetFipsCrypto(const FunctionCallbackInfo<Value>& args) {
 void SetFipsCrypto(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
 #ifdef NODE_FIPS_MODE
-  bool mode = args[0]->BooleanValue();
+  const bool enabled = FIPS_mode();
+  const bool enable = args[0]->BooleanValue();
+  if (enable == enabled)
+    return;  // No action needed.
   if (force_fips_crypto) {
```

Comment:
@jasnell if `force_fips_crypto` is on, then `FIPS_mode()` should already be enabled right? So the `if (enable == enabled)` branch would already have returned.